libsolidity/formal/Why3Translator.cpp

```diff
@@ -428,7 +428,7 @@ bool Why3Translator::visit(BinaryOperation const& _binaryOperation)
 	Type const& commonType = *_binaryOperation.annotation().commonType;
 	Token::Value const c_op = _binaryOperation.getOperator();
 
-	if (commonType.category() == Type::Category::IntegerConstant)
+	if (commonType.category() == Type::Category::NumberConstant)
```

Comment:
Please throw an error if this is not an integer (not sure if this is the correct place).